### PR TITLE
[no merge] comments

### PR DESCRIPTION
**Pull request by longbowlu**

### CODE REVIEW

File: contracts/BridgeCommittee.sol [hunk 1]
```diff
@@ -20,16 +20,21 @@ contract BridgeCommittee {
     mapping(address => uint256) public committee;
     // member address => is blocklisted
     mapping(address => bool) public blocklist;
+    // Can we use `processed` v.s. `approved`?
     // message hash => approved
-    mapping(bytes32 => bool) public messageApproved;
+    mapping(bytes32 => bool) public messageProcessed;
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:25]: agree

File: contracts/BridgeCommittee.sol [hunk 2]
```diff
@@ -20,16 +20,21 @@ contract BridgeCommittee {
...
     // message type => required amount of approval stake
+    // Nit: `requiredApprovalStake`
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:27]: yes

File: contracts/BridgeCommittee.sol [hunk 3]
```diff
@@ -52,13 +57,16 @@ contract BridgeCommittee {
         // Prepare the message hash
         bytes32 messageHash = getMessageHash(message);
         // Check that the message has not already been approved
-        require(messageApproved[messageHash], "BridgeCommittee: Message already approved");
+        // nit: `approved` => `processed`
+        require(!messageProcessed[messageHash], "BridgeCommittee: Message already approved");
 
+        // this does not match the encoding rules: `SUI_NATIVE_BRIDGE` should be prefixed to raw message instead of hash.
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:63]: Will need to explore how this should work using real messages but more than likely you're correct.

File: contracts/BridgeCommittee.sol [hunk 4]
```diff
@@ -77,16 +85,28 @@ contract BridgeCommittee {
             approvalStake += committee[signer];
         }
 
+        // Need to check the message version matches a certain number here.
+
+        // I don't quite like that we decode the same message a few times. Can we decode it once here
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:90]: lets dive into this on a call

File: contracts/BridgeCommittee.sol [hunk 5]
```diff
@@ -77,16 +85,28 @@ contract BridgeCommittee {
...
+            // 1. we already have `messageHash` from earlier
+            // 2. renaming
+            messageProcessed[messageHash] = true;
         }
     }
 
     function _processMessage(bytes memory message) private {
         uint256 messageType = decodeMessageType(message);
 
+        // For non token transfer messages, the nonce must match the next available nonce, namely we always process them one by one
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:108]: good catch

File: contracts/BridgeCommittee.sol [hunk 6]
```diff
@@ -96,29 +116,36 @@ contract BridgeCommittee {
         } else if (messageType == BRIDGE_OWNERSHIP) {
             address newOwner = decodeBridgeOwnershipMessage(message);
             _transferBridgeOwnership(newOwner);
+        // Need to check the message type is legit (e.g. can't be 10000)
         } else {
             // if the message type is not for the committee, submit it to the bridge
             bridge.submitMessage(message);
         }
-        emit MessageProcessed(message);
+        emit MessageProcessedEvent(message);
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:124]: type notation is an anti pattern

File: contracts/BridgeCommittee.sol [hunk 7]
```diff
@@ -96,29 +116,36 @@ contract BridgeCommittee {
...
     }
 
+    // Is this the regular way to upgrade?
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:127]: short answer is no. Typically upgrades are handled by a single admin signer or a multisig (that handles most of the details) I'm unclear how upgrading via another contract should work so this function is subject to change. (working on this now)

File: contracts/BridgeCommittee.sol [hunk 8]
```diff
@@ -96,29 +116,36 @@ contract BridgeCommittee {
...
+        // emit event
     }
 
     function _transferBridgeOwnership(address newOwner) internal {
         bridge.transferOwnership(newOwner);
+        // emit event
     }
 
+    // Should we check the original value, for example, if the member is already blocklisted, 
+    // then a subsequent blocklist will fail. It feels more conceptually right
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:143]: We could! does it make sense that supplying a duplicate blocklisted address should revert the transaction?

File: contracts/BridgeCommittee.sol [hunk 9]
```diff
@@ -128,20 +155,23 @@ contract BridgeCommittee {
         view
         returns (bool)
     {
-        // Get the message type from the message hash.
+        // I don't quite like we decode the same message a few times to get different fields. Can
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:158]: good point. it felt more readable but probably not the most efficient approach. will go back to the drawing board on this.

File: contracts/BridgeCommittee.sol [hunk 10]
```diff
@@ -128,20 +155,23 @@ contract BridgeCommittee {
...
+        // I don't quite like we decode the same message a few times to get different fields. Can 
+        // Get the message type from the message
         uint256 messageType = decodeMessageType(message);
         // Get the required stake for the message type
         uint256 requiredStake = requiredApproval[messageType];
         if (messageType == EMERGENCY_OP) {
             // decode the emergency op message
             bool isPausing = decodeEmergencyOpMessage(message);
             // if the message is to unpause the bridge, use the upgrade stake requirement
+            // Can we make a new constant instead of using BRIDGE_UPGRADE?
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:167]: Fair

File: contracts/BridgeCommittee.sol [hunk 11]
```diff
@@ -158,6 +188,7 @@ contract BridgeCommittee {
             abi.decode(message, (uint256, uint256, uint256, uint256));
         // 0 = pausing
         // 1 = unpausing
+        // add check: assert opCode == 0 or == 1
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:191]: on it

File: contracts/BridgeCommittee.sol [hunk 12]
```diff
@@ -128,20 +155,23 @@ contract BridgeCommittee {
...
             bool isPausing = decodeEmergencyOpMessage(message);
             // if the message is to unpause the bridge, use the upgrade stake requirement
+            // Can we make a new constant instead of using BRIDGE_UPGRADE?
             if (!isPausing) requiredStake = requiredApproval[BRIDGE_UPGRADE];
         }
         // Compare the approval stake with the required stake and return the result
         return approvalStake >= requiredStake;
     }
 
+    // Please check encoding rule, the order is type (u8) +version (u8) +nonce (u64), ditto for other functions
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:174]: not sure how this didn't translate. good catch

File: contracts/BridgeCommittee.sol [hunk 13]
```diff
@@ -186,8 +217,10 @@ contract BridgeCommittee {
             uint256 blocklistType,
             address[] memory validators
         ) = abi.decode(message, (uint256, uint256, uint256, uint256, address[]));
-        bool blocklisted = (blocklistType == 0) ? true : false;
-        return (validators, blocklisted);
+        // 0 = blocklist
+        // 1 = unblocklist
+        // add check: assert blocklistType == 0 or == 1
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:222]: done

File: contracts/BridgeCommittee.sol [hunk 14]
```diff
@@ -295,5 +329,6 @@ contract BridgeCommittee {
 
     /* ========== EVENTS ========== */
 
-    event MessageProcessed(bytes message);
+    // Can we emit human readable events?
+    event MessageProcessedEvent(bytes message);
```
Comment (Bridgerz) [inline contracts/BridgeCommittee.sol:333]: anti pattern

File: contracts/SuiBridge.sol [hunk 15]
```diff
@@ -36,18 +36,26 @@ contract SuiBridge is
 
     address[] public supportedTokens;
 
+    // So _weth9 should be 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 on mainnet?
```
Comment (Bridgerz) [inline contracts/SuiBridge.sol:39]: correct! and we'll be writing our unit tests off of mainnet

File: contracts/SuiBridge.sol [hunk 16]
```diff
@@ -36,18 +36,26 @@ contract SuiBridge is
...
+
         supportedTokens = _supportedTokens;
         vault = IBridgeVault(_vault);
         weth9 = IWETH9(_weth9);
     }
 
+    // can we have two different functions:
+    // 1. handleTokenTransfer(u8 mesasage_version, bytes memory payload) and
+    // 2. handleEmergencyOp(u8 mesasage_version, bytes memory payload) 
+    // the merit is no need for one more decoding in line 62
```
Comment (Bridgerz) [inline contracts/SuiBridge.sol:58]: No more decoding yes, but then we won't be able to introduce new messages without transferring bridge ownership to a new committee that can decode the new message type.

File: contracts/SuiBridge.sol [hunk 17]
```diff
@@ -126,11 +134,14 @@ contract SuiBridge is
             );
     }
 
+    // Same: pass the payload instead of the full message
     function _processTokenTransferMessage(bytes memory message) internal whenNotPaused {
         // Decode the message
+        // interesting, why? it does not look like a lot of data to decode
```
Comment (Bridgerz) [inline contracts/SuiBridge.sol:140]: solved for. this was the result of a late night brain 😴

File: contracts/SuiBridge.sol [hunk 18]
```diff
@@ -126,11 +134,14 @@ contract SuiBridge is
...
         // TODO: this is causing a "Stack Too Deep" error. Need to refactor
         // https://soliditydeveloper.com/stacktoodeep
         (
+            // Not all of the unit are unit256, please check the notion page
```
Comment (Bridgerz) [inline contracts/SuiBridge.sol:144]: again, haven't addressed this anywhere (focused on overall contract structure). having @georgedigkas tackle going through and altering types to align with the design in notion.

File: contracts/interfaces/IWETH9.sol [hunk 19]
```diff
@@ -4,6 +4,7 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
 /// @title Interface for WETH9
+// curious what does `9` mean?
```
Comment (Bridgerz) [inline contracts/interfaces/IWETH9.sol:7]: not sure! thats the name of THE wrapped eth contract tho...
https://etherscan.io/address/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2#code

Comment (longbowlu) [reply contracts/interfaces/IWETH9.sol:7]: probably versioning

**Pull request closed without merging**